lib/smart_answer/calculators/check_travel_during_coronavirus_calculator.rb

```diff
@@ -22,6 +22,15 @@ def country_locations
       end
     end
 
+    def summary_text_fields
+      return [] if travelling_to_ireland? && single_journey?
+
+      @summary_text_fields = @travelling_with_children.dup
+      @summary_text_fields << "young_person" if travelling_with_young_people?
+      @summary_text_fields << "red_list" if travelling_to_red_list_country?
+      @summary_text_fields << (vaccination_status == "9ddc7655bfd0d477" ? "not_vaxed" : "fully_vaxed")
+    end
```

Comment:
So let's see if I understand correctly. You are creating an array called `@summary_text_fields` (why is it an instance variable?)

And in `@summary_text_fields` you're adding codes to look up the text value of what should be displayed in the "Because you said" column?

I think this would be much easier if you started with populating the text here directly in this commit and then try moving to a config file, as at that point you'd have working tests to make sure you're plucking out the correct thing.

Regardless I still think a few `if/else` statements would make things easier. For example:

```
if travelling_with_children?
  @summary_text_field << travelling_with_children
elsif travelling_with_young_people?
  @summary_text_field << "young_person"
end

if vaccination_status == "9ddc7655bfd0d477"
  @summary_text_field << "not_vaxed"
elsif vaccination_status == "code for fully vaxed" || vaccination_status == "code for vaccination trial"
   @summary_text_field << "fully_vaxed"
end

@summary_text_fields << "red_list" if travelling_to_red_list_country?
```
etc

I think `travelling_with_children` returns an array, so you may need to return `@summary_text_fields.flatten` on the last line to make it work.

As mentioned before I'd also recommend adding a method here that takes the result of `summary_text_fields`  and uses it to look up the long text from the YAML file, and return a list that can be looped through in the view.

Reply:
Actually, my brain seems to working better this morning. And I have a better suggestion. Rather than creating a list of "codes" i.e. ["fully_vaxed", "young_person"], why not create a method that loads the YAML and lets you look up the text string by code. For example, if you have a method called `def find_summary_text_from_code(code)` then if the `summary_text_fields` you could call:

```
if vaccination_status == "9ddc7655bfd0d477"
  @summary_text_field << find_summary_text_from_code("not_vaxed")
elsif vaccination_status == "code for fully vaxed" || vaccination_status == "code for vaccination trial"
   @summary_text_field << find_summary_text_from_code("fully_vaxed")
end
```

Reply:
Again, sorted by internationalising :)